src/OpenTelemetry.Exporter.Prometheus/Implementation/PrometheusExporterMetricsHttpServer.cs

```diff
@@ -128,13 +128,49 @@ private void WorkerThread()
                     ctxTask.Wait(this.tokenSource.Token);
                     var ctx = ctxTask.Result;
 
-                    if (!this.exporter.TryEnterSemaphore())
+                    try
                     {
-                        ctx.Response.StatusCode = 429;
-                        ctx.Response.Close();
+                        ctx.Response.StatusCode = 200;
+                        ctx.Response.ContentType = PrometheusMetricsFormatHelper.ContentType;
+
+                        this.exporter.OnExport = (metrics) =>
+                        {
+                            try
+                            {
+                                var buffer = new byte[65536];
```

Comment:
This should be statically allocated.
Need to catch index out of range error and reallocate.